js/components/MilkdropWindow/index.js

```diff
@@ -164,12 +178,27 @@ async function loadInitialPreset() {
         throw new Error(response.statusText);
       }
       const responseBody = await response.json();
-      return responseBody;
+      const presetName = presetNameFromURL(presetUrl);
+      return { [presetName]: responseBody };
     } catch (e) {
       console.error(e);
       alert(`Failed to load MilkDrop preset from ${presetUrl}`);
       return null;
     }
+  } else if (milkdropPresetUrl) {
+    try {
+      const response = await fetch(milkdropPresetUrl);
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      const fileBlob = await response.blob();
+      const presetName = presetNameFromURL(milkdropPresetUrl);
+      return { [presetName]: { file: fileBlob } };
+    } catch (e) {
+      console.error(e);
+      alert(`Failed to load MilkDrop preset from ${milkdropPresetUrl}`);
```

Comment:
@durasj Hey, how do you feel about us using alerts to notify users of failures? Would it be better for us to have some internal "notification" api which defaults to alert, but could be optionally injected by the consumer? I'm thinking that there might be a better way to handle these kinds of things in the Electron app.